test/api/tracker/api_test.go

```diff
@@ -0,0 +1,192 @@
+package tracker
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/konveyor/tackle2-hub/api"
+	"github.com/konveyor/tackle2-hub/test/assert"
+)
+
+func TestTrackerCRUD(t *testing.T) {
+	for _, r := range Samples {
+		t.Run("Tracker CRUD", func(t *testing.T) {
+			// Create a sample identity for the tracker.
+			identity := api.Identity{
+				Kind: r.Kind,
+				Name: r.Identity.Name,
+			}
+			assert.Must(t, Identity.Create(&identity))
+			// Copy the identity name to the tracker.
+			r.Identity.ID = identity.ID
+
+			// Create a tracker.
+			assert.Must(t, Tracker.Create(&r))
+
+			// Get.
+			got, err := Tracker.Get(r.ID)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Compare got values with expected values.
+			AssertEqualTrackers(t, got, r)
+
+			// Update Message.
+			r.Message = "Updated Comment " + r.Message
+			assert.Should(t, Tracker.Update(&r))
+
+			// Find Tracker and check its parameters with the got(On Updation).
+			got, err = Tracker.Get(r.ID)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Check if the unchanged values remain same or not.
+			AssertEqualTrackers(t, got, r)
+
+			// Delete identity and tracker.
+			assert.Must(t, Tracker.Delete(r.ID))
+			assert.Must(t, Identity.Delete(identity.ID))
+
+			// Check if the Tracker is present even after deletion or not.
+			_, err = Tracker.Get(r.ID)
+			if err == nil {
+				t.Errorf("Resource exits, but should be deleted: %v", r)
+			}
+		})
+
+		t.Run("Tracker Project", func(t *testing.T) {
+			// Create a sample identity for the tracker.
+			identity := api.Identity{
+				Kind: r.Kind,
+				Name: r.Identity.Name,
+			}
+			assert.Must(t, Identity.Create(&identity))
+
+			// Copy the identity name to the tracker.
+			r.Identity.ID = identity.ID
+
+			// Create a tracker.
+			assert.Must(t, Tracker.Create(&r))
+
+			projectsList, err := Tracker.ListProjects(r.ID)
+			if err != nil {
+				// check for type of service (maybe Connected?)
+				// if API service then print Not connected to Jira and pass the test
+				// else fail the test
+				if r.Connected == false {
+					t.Logf("Not connected to Jira(Thus passing the API Test)")
+				}
+				if r.Connected == true {
```

Comment:
Thinking why not use ```} else {``` here (or do you want extent the condition with more checks?)

Reply:
No I was thinking if this condition would suffice or if we would need more conditions later. Will update.Thanx